cumulusci/robotframework/Salesforce.py

```diff
@@ -448,35 +380,24 @@ def store_session_record(self, obj_type, obj_id):
         self.builtin.log("Storing {} {} to session records".format(obj_type, obj_id))
         self._session_records.append({"type": obj_type, "id": obj_id})
 
+    @selenium_retry
     def wait_until_modal_is_open(self):
-        """ EXPERIMENTAL!!! """
-        self._call_selenium("_wait_until_modal_is_open", True)
-
-    def _wait_until_modal_is_open(self):
-        self.selenium.wait_until_element_is_not_visible(
-            lex_locators["spinner"], timeout=15
-        )
-        self.selenium.wait_until_element_is_visible(
+        """ Wait for modal to open """
+        self.selenium.wait_until_page_contains_element(
             lex_locators["modal"]["is_open"], timeout=15
         )
 
+    @selenium_retry
     def wait_until_modal_is_closed(self):
-        """ EXPERIMENTAL!!! """
-        self._call_selenium("_wait_until_modal_is_closed", True)
-
-    def _wait_until_modal_is_closed(self):
-        self.selenium.wait_until_element_is_not_visible(
-            lex_locators["spinner"], timeout=15
-        )
-        self.selenium.wait_until_element_is_not_visible(
+        """ Wait for modal to close """
+        self.selenium.wait_until_page_does_not_contain_element(
             lex_locators["modal"]["is_open"], timeout=15
         )
 
     def wait_until_loading_is_complete(self):
-        """ EXPERIMENTAL!!! """
-        self._call_selenium("_wait_for_aura", True)
-
+        """ Wait until Aura has processed all in flight XHTTP requests """
```

Comment:
is this when aura has finished *processing* any XHRs or just the XHR itself is finished? has the callback been executed?

Reply:
I think just the xhr itself. I spent quite a while trying to figure out how to wait for all processing to be done but Aura doesn't expose the necessary hooks (it's fairly complicated given that processing the result of an xhr can trigger further xhrs...we basically need a hook from the event loop saying that it's idle).

Reply:
can you synthesize that into something presentable for the public docstring?